include/eve/concept/scalar.hpp

```diff
@@ -12,39 +12,98 @@
 
 #include <cstddef>
 #include <cstdint>
-
 #include <type_traits>
 
-namespace eve
+namespace eve::detail
 {
-namespace detail
+template<typename T>
+constexpr bool is_plain() noexcept
 {
-  template<typename T>
-  constexpr bool is_plain() noexcept
-  {
-    return !(std::is_same_v<T, bool> || std::is_same_v<T, long double>)
+  return    !(std::is_same_v<T, bool> || std::is_same_v<T, long double>)
         &&  (std::is_floating_point_v<T> || std::is_integral_v<T>);
-  }
+}
```

Comment:
this should be a concept I think. And don't we have it already?

Reply:
That's the trait we use to define the concept. It has been implemented and validated when we merged the last concept revision.